scalagen-sbt/src/main/scala/scala/meta/gen/ScalagenPlugin.scala

```diff
@@ -0,0 +1,85 @@
+package scala.meta.gen
+
+import java.io.File
+import java.nio.file.Path
+
+import org.scalameta.scalagen.Runner
+import sbt._
+import sbt.Keys._
+
+import scala.meta._
+
+object ScalagenTags {
+  val SourceGeneration = Tags.Tag("source-generation")
+}
+
+object ScalagenPlugin extends AutoPlugin {
+  override def trigger = allRequirements
+
+  object autoImport {
+
+    lazy val listGenerators =
```

Comment:
sbt plugins conventionally prefix tasks with the plugin name, so this task should be named `scalagenListGenerators`. This convention has some nice benefits, like easier explorability with tab completions inthe shell